acceptance-tests/src/main/java/io/blueocean/ath/model/Folder.java

```diff
@@ -40,7 +40,7 @@ public String get(int i) {
 
     public Folder append(String ...folders) {
         List<String> newFolders = Stream.concat(this.folders.stream(), Arrays.stream(folders)).collect(Collectors.toList());
-        return new Folder(ImmutableList.copyOf(newFolders));
+        return new Folder( Collections.unmodifiableList(new ArrayList<>( newFolders)));
```

Comment:
I didn't see this before, but it just got a bit weird :)
Here you create a new list from the streams of the existing list + the new list.
You then take that newly created list and create a new `ArrayList` with those elements and wrap that in an unmodifiable list, that is then passed to the constructor that again creates a new `ArrayList` which is modifiable.

A lot of Lists and wrappings created and elements copied over and over just to end up again with a plain old list :)

Reply:
OMG yes totally wrong and for something used only in a test module...
I will simplify that